components/AppContext.tsx

```diff
@@ -0,0 +1,11 @@
+import { Color } from '@material-ui/core'
+import { createContext } from 'react'
+
+interface AppContextStructure {
+	themeType: 'dark' | 'light'
+	setThemeType: (type: string) => void
```

Comment:
themeType: PaletteType
	setThemeType: (type: PaletteType) => void

would be DRYer, since `@material-ui/core` exports `type PaletteType = 'light' | 'dark'`